packages/odyssey-react-mui/src/labs/DataComponents/StackCard.tsx

```diff
@@ -0,0 +1,194 @@
+import {
+  DesignTokens,
+  useOdysseyDesignTokens,
+} from "../../OdysseyDesignTokensContext";
+import { Heading5, Paragraph, Support } from "../../Typography";
+import { Box } from "../../Box";
+
+export const CARD_IMAGE_HEIGHT = "64px";
+
+export type StackCardProps = {
+  description?: string;
+  image?: ReactElement;
+  overline?: string;
+  title?: string;
+  children?: ReactNode;
+} & (
+  | {
+      onClick: MouseEventHandler;
+      button?: never;
+      menuButtonChildren?: never;
+      Accessory?: never;
+    }
+  | {
+      onClick?: never;
+      button?: ReactElement<typeof Button>;
+      menuButtonChildren?: MenuButtonProps["children"];
+      Accessory?: ReactNode;
+    }
+);
+
+const ImageContainer = styled("div", {
+  shouldForwardProp: (prop) =>
+    prop !== "odysseyDesignTokens" && prop !== "hasMenuButtonChildren",
+})<{
+  odysseyDesignTokens: DesignTokens;
+  hasMenuButtonChildren: boolean;
+}>(({ odysseyDesignTokens, hasMenuButtonChildren }) => ({
+  display: "flex",
+  alignItems: "flex-start",
+  maxHeight: `${CARD_IMAGE_HEIGHT}`,
+  marginBlockEnd: odysseyDesignTokens.Spacing5,
+  paddingRight: hasMenuButtonChildren ? odysseyDesignTokens.Spacing5 : 0,
+}));
+
+const MenuButtonContainer = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{ odysseyDesignTokens: DesignTokens }>(({ odysseyDesignTokens }) => ({
+  position: "absolute",
+  right: odysseyDesignTokens.Spacing3,
+  top: odysseyDesignTokens.Spacing3,
+}));
+
+const buttonProviderValue = { isFullWidth: true };
+
+const StackCard = ({
+  button,
+  description,
+  image,
+  menuButtonChildren,
+  onClick,
+  overline,
+  title,
+  children,
+  Accessory,
+}: StackCardProps) => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  const cardContent = useMemo(
+    () => (
+      <Box
+        sx={{
+          display: "flex",
+        }}
+      >
+        {Accessory && (
+          <Box
+            sx={{
+              marginInlineEnd: 3,
+              marginInlineStart: -2,
+            }}
+          >
+            {Accessory}
+          </Box>
+        )}
+        <Box>
+          {image && (
+            <ImageContainer
+              odysseyDesignTokens={odysseyDesignTokens}
+              hasMenuButtonChildren={Boolean(menuButtonChildren)}
+            >
+              {image}
+            </ImageContainer>
+          )}
+
+          {overline && <Support component="div">{overline}</Support>}
+          {title && <Heading5 component="div">{title}</Heading5>}
+          {description && (
+            <Paragraph color="textSecondary">{description}</Paragraph>
+          )}
+
+          {button && (
+            <MuiCardActions>
+              <ButtonContext.Provider value={buttonProviderValue}>
+                {button}
+              </ButtonContext.Provider>
+            </MuiCardActions>
+          )}
+
+          {children && (
+            <Box
+              sx={{
+                ["&:not(:first-child)"]: {
```

Comment:
My sense is it's helpful to use styled-components when practical (doesn't require major rework or hacking in additional elements) / when it will help others update the file in the future.  

For something like this ` <Box sx={{ width: tableInnerContainerWidth }}>{emptyState}</Box>,` (from further down in the code) using `sx` helps me read the code because of its proximity to where the variable is defined. For adding tokens and box model I prefer styled components though. 



TL;DR My opinion is to use styled-components when practical and `sx` should be the exception, not the rule.